cmake/manifoldDeps.cmake

```diff
@@ -0,0 +1,226 @@
+# limitations under the License.
+include(FetchContent)
+
+function(logmissingdep PKG)
+  # if this is already in FETCHCONTENT_SOURCE_DIR_X, we don't have to
+  # download...
+  if(DEFINED FETCHCONTENT_SOURCE_DIR_${PKG})
+    return()
+  endif()
+  if(NOT MANIFOLD_DOWNLOADS)
+    if(ARGC EQUAL 3)
+      set(MSG "${ARGV2} enabled, but ${PKG} was not found ")
+      string(APPEND MSG "and dependency downloading is disabled. ")
+    else()
+      set(MSG "${PKG} not found, and dependency downloading disabled. ")
+    endif()
+    string(APPEND MSG "Please install ${PKG} and reconfigure.\n")
+    message(FATAL_ERROR ${MSG})
+  else()
+    message(STATUS "${PKG} not found, downloading from source")
+  endif()
+endfunction()
+
+set(OLD_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
+# we build fetched dependencies as static library
+set(BUILD_SHARED_LIBS OFF)
+
+# If we're building parallel, we need the requisite libraries
+set(MANIFOLD_BUILTIN_TBB OFF)
+if(MANIFOLD_PAR)
+  find_package(Threads REQUIRED)
+  find_package(TBB QUIET)
+  find_package(PkgConfig QUIET)
+  if(NOT TBB_FOUND AND PKG_CONFIG_FOUND)
+    pkg_check_modules(TBB tbb)
+  endif()
+  if(NOT TBB_FOUND)
+    logmissingdep("TBB" , "Parallel mode")
+    set(TBB_TEST OFF CACHE INTERNAL "" FORCE)
+    set(TBB_STRICT OFF CACHE INTERNAL "" FORCE)
+    FetchContent_Declare(
+      TBB
+      GIT_REPOSITORY https://github.com/oneapi-src/oneTBB.git
+      GIT_TAG v2021.11.0
+      GIT_PROGRESS TRUE
+      EXCLUDE_FROM_ALL
+    )
+    FetchContent_MakeAvailable(TBB)
+    set(MANIFOLD_BUILTIN_TBB ON)
+  endif()
+  if(NOT TARGET TBB::tbb)
+    if(NOT TARGET tbb)
+      add_library(TBB::tbb SHARED IMPORTED)
+      set_property(
+        TARGET TBB::tbb
+        PROPERTY IMPORTED_LOCATION ${TBB_LINK_LIBRARIES}
+      )
+      target_include_directories(TBB::tbb INTERFACE ${TBB_INCLUDE_DIRS})
+    else()
+      add_library(TBB::tbb ALIAS tbb)
+    endif()
+  endif()
+endif()
+
+# If we're building cross_section, we need Clipper2
+set(MANIFOLD_BUILTIN_CLIPPER2 OFF)
+if(MANIFOLD_CROSS_SECTION)
+  find_package(Clipper2 QUIET)
+  if(NOT Clipper2_FOUND AND PKG_CONFIG_FOUND)
+    pkg_check_modules(Clipper2 Clipper2)
+  endif()
+  if(Clipper2_FOUND)
+    add_library(Clipper2 SHARED IMPORTED)
+    set_property(
+      TARGET Clipper2
+      PROPERTY IMPORTED_LOCATION ${Clipper2_LINK_LIBRARIES}
+    )
+    if(WIN32)
+      set_property(
+        TARGET Clipper2
+        PROPERTY IMPORTED_IMPLIB ${Clipper2_LINK_LIBRARIES}
+      )
+    endif()
+    target_include_directories(Clipper2 INTERFACE ${Clipper2_INCLUDE_DIRS})
+  else()
+    logmissingdep("Clipper2" , "cross_section")
+    set(MANIFOLD_BUILTIN_CLIPPER2 ON)
+    set(CLIPPER2_UTILS OFF)
+    set(CLIPPER2_EXAMPLES OFF)
+    set(CLIPPER2_TESTS OFF)
+    set(
+      CLIPPER2_USINGZ
+      OFF
+      CACHE BOOL
+      "Preempt cache default of USINGZ (we only use 2d)"
+    )
+    FetchContent_Declare(
+      Clipper2
```

Comment:
Does this mean we're now auto-fetching dependencies again? How does this relate to our CMake flag for fetching dependencies?

Reply:
No, the `logmissingdep` function check the cmake flag and will error when we disabled download.